cli/cli.go

```diff
@@ -124,15 +123,19 @@ func parseFormatString(arg string) (feedback.OutputFormat, bool) {
 func searchConfigTree(cwd string) string {
 	// go back up to root and search for the config file
 	for {
-		if _, err := os.Stat(path.Join(cwd, "arduino-cli.yaml")); os.IsNotExist(err) {
+		if _, err := os.Stat(filepath.Join(cwd, "arduino-cli.yaml")); err == nil {
+			// config file found
+			return cwd
+		} else if os.IsNotExist(err) {
 			// no config file found
-			next := path.Join(cwd, "..")
-			if filepath.Clean(next) == filepath.Clean(cwd) {
```

Comment:
After a quick profiling session it seems that on Windows  the `filepath.Clean()` was responsible for the performance degradation. Removing this check fixed the CLI slow response (even with the `version` command) in case of missing `arduino-cli.yaml` configuration file on the machine.